packages/react-native/ReactCommon/react/renderer/components/modal/ModalHostViewState.h

```diff
@@ -25,7 +29,12 @@ class ModalHostViewState final {
  public:
   using Shared = std::shared_ptr<const ModalHostViewState>;
 
-  ModalHostViewState(){};
+#if defined(__APPLE__) && TARGET_OS_IOS
+  ModalHostViewState() : screenSize(RCTModalHostViewScreenSize()) {
+#else
+  ModalHostViewState(){
+#endif
+  };
```

Comment:
@cipolleschi  I tried to change the size before the mount view, but it's too late because we need to get the correct layout info in the layout phase https://github.com/facebook/react-native/blob/1d8dea54b6ec0f3bb56cbd64c4d68a743fe4150f/packages/react-native/ReactCommon/react/renderer/components/modal/ModalHostViewComponentDescriptor.h#L33